pkg/vm/stackitem/serialization.go

```diff
@@ -237,26 +245,35 @@ func Deserialize(data []byte) (Item, error) {
 // as a function because Item itself is an interface. Caveat: always check
 // reader's error value before using the returned Item.
 func DecodeBinary(r *io.BinReader) Item {
-	dc := deserContext{
-		BinReader:    r,
-		allowInvalid: false,
-		limit:        MaxDeserialized,
-	}
-	return dc.decodeBinary()
+	dc := NewDeserializationContext()
+	return dc.Deserialize(r, false, MaxDeserialized)
 }
 
 // DecodeBinaryProtected is similar to DecodeBinary but allows Interop and
 // Invalid values to be present (making it symmetric to EncodeBinaryProtected).
 func DecodeBinaryProtected(r *io.BinReader) Item {
-	dc := deserContext{
-		BinReader:    r,
-		allowInvalid: true,
-		limit:        MaxDeserialized,
+	dc := NewDeserializationContext()
+	return dc.Deserialize(r, true, MaxDeserialized)
+}
+
+// Deserialize returns Item deserialized from the given BinReader. allowInvalid
+// denotes whether it is allowed for Interop items and items of InvalidT type to
+// be present in the resulting stackitem. limit restricts the maximum number of
+// items deserialized item can contain (including itself). The default limit of
+// MaxDeserialized is used if non-positive limit is specified. The resulting
+// context's Err must be checked after deserialization, it is valid until the next
+// call to Deserialize.
+func (r *DeserializationContext) Deserialize(reader *io.BinReader, allowInvalid bool, limit int) Item {
```

Comment:
So this might as well be function, not a method, you're completely defining the context in parameters.

Reply:
Fixed.